cirq-core/cirq/circuits/circuit_test.py

```diff
@@ -254,6 +253,51 @@ def test_append_control_key():
     assert len(c) == 1
 
 
+def test_control_key_diagram():
```

Comment:
Add some more tests, covering each of the bullet points in the PR, plus subcircuits.